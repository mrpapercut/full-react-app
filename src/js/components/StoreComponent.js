import {Component} from 'react';
import {connect} from 'react-redux';

import createElement from '~/util/createElement';

const [div, h2] = ['div', 'h2'].map(createElement);

class StoreComponent extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return div({
            className: 'storecomponent'
        },
            h2({
                className: 'pagetitle'
            }, 'Store Component')
        );
    }
}

const mapStateToProps = ({mainReducer}) => {
    return {
        isLoading: mainReducer.isLoading
    };
};

const mapDispatchToProps = dispatch => {
    return {
        setLoading: bool => dispatch(isLoading(bool))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(StoreComponent);
