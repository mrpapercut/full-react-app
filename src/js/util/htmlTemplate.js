const htmlTemplate = ({title, body, scriptTags, preloadedState}) => `<!doctype html>
<html>
<head>
    <meta charset="utf8"></meta>
    <title>${title}</title>
    <base href="/">

    <meta name="viewport" content="width=device-width, initial-scale=1.0">
</head>
<body>
    <div id="appwrapper">${body}</div>
    <script>${preloadedState}</script>
    ${scriptTags}
</body>
</html>
`;

export default htmlTemplate;
