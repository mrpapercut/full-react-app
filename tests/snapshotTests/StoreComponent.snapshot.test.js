import {createElement} from 'react';
import renderer from 'react-test-renderer';

import {wrapElementWithRouter} from '../util/snapshot-test-util';

import StoreComponent from '~/components/StoreComponent';

test('StoreComponent', () => {
    const component = renderer.create(wrapElementWithRouter(createElement(StoreComponent)));
    expect(component.toJSON()).toMatchSnapshot();
});
