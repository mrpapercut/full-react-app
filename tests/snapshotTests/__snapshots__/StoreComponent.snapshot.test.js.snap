// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreComponent 1`] = `
<div
  className="storecomponent"
>
  <h2
    className="pagetitle"
  >
    Store Component
  </h2>
</div>
`;
